### PR TITLE
Remove use of mattn/go-shellwords

### PR DESCRIPTION
**Pull request by thaJeztah**
This dependency was only used in two locations, and didn't appear to be really needed for how it was used.

supersedes https://github.com/moby/moby/pull/41166
closes https://github.com/moby/moby/pull/41166

### CODE REVIEW

File: pkg/parsers/operatingsystem/operatingsystem_linux.go [hunk 1]
```diff
@@ -63,14 +61,7 @@ func getValueFromOsRelease(key string) (string, error) {
...
 			data := strings.SplitN(line, "=", 2)
-			values, err := shellwords.Parse(data[1])
-			if err != nil {
-				return "", fmt.Errorf("%s is invalid: %s", key, err.Error())
-			}
-			if len(values) != 1 {
-				return "", fmt.Errorf("%s needs to be enclosed by quotes if they have spaces: %s", key, data[1])
-			}
-			value = values[0]
+			value = strings.Trim(data[1], `"' `) // remove leading/trailing quotes and whitespace
```
Comment (thaJeztah) [inline pkg/parsers/operatingsystem/operatingsystem_linux.go:64]: Taking a slight shortcut here, but I don't expect this to cause problems for most cases. I can make it more strict it this is a concern

**Comment by thaJeztah** [comment]
@tonistiigi @cpuguy83 PTAL

**Comment by AkihiroSuda** [comment]
Where are we using this? dockerd doesn't support macOS

**Comment by thaJeztah** [comment]
Don't think we use it ourselves, but perhaps someone else is; it was added in https://github.com/moby/moby/pull/22549

**Review by AkihiroSuda — APPROVED** [review]

**Review by tonistiigi — APPROVED** [review]

**Comment by thaJeztah** [comment]
Failure looks unrelated (flaky test); this one good to go?

**Pull request merged**